File: day1.hs
{-# LANGUAGE OverloadedStrings #-}

import Data.List (sort)
import Data.List.Split (splitOn)

main :: IO ()
main = do
  input <- readFile "input/day1.txt"
  let totals = map (sum . map read) . splitOn [""] . lines $ input
      pt1 = foldl max 0 totals
      pt2 = sum $ take 3 $ reverse $ sort totals
  print [pt1, pt2]

File: clj/day1.clj
(ns user (:require [clojure.string :as str]))

(println
 (->> (slurp "input/day1.txt")
      str/split-lines
      (partition-by str/blank?)
      (map #(reduce (fn [a x] (+ a (or (parse-long x) 0))) 0 %))
      (apply max))

 (->> (slurp "input/day1.txt")
      str/split-lines
      (partition-by str/blank?)
      (map #(reduce (fn [a x] (+ a (or (parse-long x) 0))) 0 %))
      sort
      (take-last 3)
      (apply +)))
